client/components/buttons/DeleteButton.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import swal from 'sweetalert';
+
+
+const propTypes = {
+  handleDeleteRecipe: PropTypes.func.isRequired,
+  id: PropTypes.number.isRequired
+};
+
+class DeleteButton extends Component {
+  constructor(props) {
+    super(props);
+
+    this.onClick = this.onClick.bind(this);
```

Comment:
Add the [transform class properties plugin](https://babeljs.io/docs/plugins/transform-class-properties/) to your `.babelrc` file and you will not need to bind methods like this in the constructor. Once it's added, you can just rewrite the `onClick` method as an arrow function and it will be automatically bound to the class instance.